ideas/apq.h

```diff
@@ -0,0 +1,285 @@
+using io_service = boost::asio::io_service;
+
+
+/**
+* Should we use defined type of pg_connect_handle, or some customized type and
+* template for connection?
+*   PGconn                       - pointer to native connection handle
+*   std::function<void(PGconn*)> - finalization function, e.g. destructor for single 
+*                                  connection or recycling operation for pool.
+*/
+using pg_connect_handle = std::shared_ptr<PGconn>; //, std::function<void(PGconn*)>>;
```

Comment:
I think we definitely shouldn't use ```std::function``` for finalization. For both standalone and polled connection we know finalization functions. Let's use some customized type or two different types.

Reply:
There is a trade-off between std::function (or any other abstraction) and several types. It is about connection class. In case of the different types we need to use a template of the connection class. And this leads to templating of the transaction_connection class, commit and rollback functions. IMO this fact makes code slightly more complicated. But I can be wrong.